ports/libtorch/portfile.cmake

```
@@ -0,0 +1,187 @@
+vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO pytorch/pytorch
+    REF v1.10.0
+    SHA512 92b70e6170a7f173c4a9cb29f6cec6dfa598587aa9cf6a620ec861b95da6ea555cbc7285914c0dab6cfc8af320fad4999be4a788acc1f15140664a67ad9dc35d
+    HEAD_REF master
+    PATCHES
+        fix-cmake.patch
+        fix-sources.patch
+        use-glog-header.patch
+)
+file(REMOVE_RECURSE "${SOURCE_PATH}/caffe2/core/macros.h") # We must use generated header files
+
+vcpkg_find_acquire_program(PYTHON3)
+get_filename_component(PYTHON3_DIR "${PYTHON3}" DIRECTORY)
+vcpkg_add_to_path(PREPEND "${PYTHON3_DIR}")
+
+# run `pip install ${package}`
+function(pip_install)
+    cmake_parse_arguments(PARSE_ARGV 0 "arg" "" "PACKAGE" "INSTALL_OPTIONS")
+    vcpkg_execute_required_process(
+        COMMAND "${PYTHON3}" -m pip install ${arg_PACKAGE} ${arg_INSTALL_OPTIONS}
+        WORKING_DIRECTORY ${CURRENT_BUILDTREES_DIR}
+        LOGNAME install-${arg_PACKAGE}
+        ERROR_VARIABLE INSTALL_RETURN_CODE
+    )
+    if(INSTALL_RETURN_CODE) # non-zero means failure
+        message(FATAL_ERROR "PIP install failed. Check ${CURRENT_BUILDTREES_DIR}/install-${arg_PACKAGE}-err.log")
+    endif()
+endfunction()
+pip_install(PACKAGE typing-extensions)
+pip_install(PACKAGE pyyaml)
```

This should probably use [x_vcpkg_get_python_packages](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/ports/vcpkg-get-python-packages/x_vcpkg_get_python_packages.md) to ensure the correct Python is being used and the default python environment is not being polluted with extra packages.

eg:
```suggestion
x_vcpkg_get_python_packages(
    PYTHON_VERSION 3
    PACKAGES typing-extension pyyaml
    OUT_PYTHON_VAR PYTHON3
)
```